python/tk_multi_publish2/dialog.py

```diff
@@ -297,8 +300,18 @@ def _update_item_thumbnail(self, pixmap):
         thumbnail pixmap
         """
         if not self._current_item:
-            raise TankError("No current item set!")
-        self._current_item.thumbnail = pixmap
+            # this is the summary item
+            self._summary_thumbnail = pixmap
+            if pixmap != None:
+               # update all items with the summary thumbnail
+               for top_level_item in self._plugin_manager.top_level_items:
+                   top_level_item.thumbnail = self._summary_thumbnail
+                   top_level_item._set_thumbnail_enabled(True)
+                   top_level_item._thumbnail_overrides_summary = False
+        else:
+            self._current_item.thumbnail = pixmap
+            # specify that the new thumbnail overrides the one inherited from summary
+            self._current_item._thumbnail_overrides_summary = True
```

Comment:
We shouldn't be using protected members of an object. I'd suggest some property to indicate that a thumbnail has been explicitly set on the item instead of indicating that it should be overridden.